include/arch/x86/arch/machine/fpu.h

```diff
@@ -114,7 +114,7 @@ static inline void disableFpu(void)
 #ifdef CONFIG_VTX
 static inline bool_t vcpuThreadUsingFPU(tcb_t *thread)
 {
-    return thread->tcbArch.tcbVCPU && &thread->tcbArch.tcbVCPU->fpuState == NODE_STATE(ksActiveFPUState);
+    return thread->tcbArch.tcbVCPU && thread == NODE_STATE(ksActiveFPUThread);
```

Comment:
So this line is currently incorrect, and probably the whole reason why `ksActiveFPUState` wasn't a TCB to begin with.

Reply:
It replaced a check of `thread->tcbArch.tcbVCPU->fpuState` with an (implicit) check of `thread->tcbArch.tcbContext.fpuState`.

On x86 the thread switches between VM monitor and VM guest mode, both having their own FPU state, which we shouldn't mix up by accident.